File: src/main/java/com/example/board/util/FileUtil.java
package com.example.board.util;
import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;
import jakarta.servlet.http.HttpServletRequest;

/**
 * 파일 유틸리티 개선.
 */
public class FileUtil {

    public static byte[] readFile(HttpServletRequest request) {
        try {
            String path = request.getParameter("path");
            if (path == null || path.contains("..")) {
                throw new SecurityException("Invalid file path");
            }

            File f = new File(path);
            if (!f.exists() || !f.getCanonicalPath().startsWith("/secure-base-dir")) {
                throw new SecurityException("Access denied for file path");
            }

            try (FileInputStream fis = new FileInputStream(f)) {
                return fis.readAllBytes();
            }
        } catch (IOException e) {
            throw new RuntimeException("Error reading file", e);
        }
    }

    public static byte[] readFileByParam(HttpServletRequest request) throws IOException {
        String uploadPath = request.getParameter("uploadPath");
        if (uploadPath == null || uploadPath.contains("..")) {
            throw new SecurityException("Invalid file path");
        }

        File file = new File(uploadPath);
        if (!file.exists() || !file.getCanonicalPath().startsWith("/secure-base-dir")) {
            throw new SecurityException("Access denied for file path");
        }

        try (FileInputStream fis = new FileInputStream(file)) {
            return fis.readAllBytes();
        }
    }
}

File: src/main/java/com/example/board/util/FileUtil.py
# File: utils/file_util.py

import os
from fastapi import HTTPException, Request

class FileUtil:
    @staticmethod
    async def read_file(request: Request) -> bytes:
        try:
            path = request.query_params.get("path")
            if path is None or ".." in path:
                raise HTTPException(status_code=400, detail="Invalid file path.")
            
            # Validate file path
            canonical_path = os.path.realpath(path)
            if not os.path.exists(path) or not canonical_path.startswith("/secure-base-dir"):
                raise HTTPException(status_code=403, detail="Access denied for file path.")

            # Read the file content
            with open(canonical_path, "rb") as file:
                return file.read()
        except Exception as e:
            raise HTTPException(status_code=500, detail=f"Error reading file: {str(e)}")

    @staticmethod
    async def read_file_by_param(request: Request) -> bytes:
        try:
            upload_path = request.query_params.get("uploadPath")
            if upload_path is None or ".." in upload_path:
                raise HTTPException(status_code=400, detail="Invalid file path.")
            
            # Validate file path
            canonical_path = os.path.realpath(upload_path)
            if not os.path.exists(upload_path) or not canonical_path.startswith("/secure-base-dir"):
                raise HTTPException(status_code=403, detail="Access denied for file path.")

            # Read the file content
            with open(canonical_path, "rb") as file:
                return file.read()
        except Exception as e:
            raise HTTPException(status_code=500, detail=f"Error reading file: {str(e)}")